clients/baseclient/client_util.go

```diff
@@ -23,7 +23,11 @@ func shouldRetry(err error) bool {
 	if err == nil {
 		return false
 	}
-	isMatching, _ := regexp.MatchString(" EOF$", err.Error())
+	isMatching := strings.Contains(err.Error(), "retry is needed")
```

Comment:
Maybe the 2 checks could be extracted in a separate methods for better readability?

Reply:
Refactored -> new method

Reply:
Do not see this new method.